test/Dialect/FIRRTL/imconstprop.mlir

```diff
@@ -542,3 +542,198 @@ firrtl.circuit "dntOutput" {
+    %3 = firrtl.xor %0, %1 : (!firrtl.uint<3>, !firrtl.uint<3>) -> !firrtl.uint<3>
+    %4 = firrtl.xor %3, %2 : (!firrtl.uint<3>, !firrtl.uint<3>) -> !firrtl.uint<3>
+    firrtl.connect %result, %4 : !firrtl.uint<3>, !firrtl.uint<3>
+    // TODO: Register and connections are not currently erased.
+    // CHECK: %tmp = firrtl.reg %clock
+
+    // CHECK:      %c7_ui3 = firrtl.constant 7 : !firrtl.uint<3>
+    // CHECK-NEXT: firrtl.connect %result, %c7_ui3 : !firrtl.uint<3>, !firrtl.uint<3>
+  }
+}
+
+// -----
+
+firrtl.circuit "VectorOfBundlePropagation"   {
+  // CHECK-LABEL: @VectorOfBundlePropagation
+  firrtl.module @VectorOfBundlePropagation(in %clock: !firrtl.clock, out %res1: !firrtl.uint<3>, out %res2: !firrtl.uint<3>) {
+    %bundle_vec = firrtl.reg %clock  : !firrtl.vector<bundle<a: uint<3>, b: uint<3>>, 3>
+    %invalid = firrtl.invalidvalue : !firrtl.vector<bundle<a: uint<3>, b: uint<3>>, 3>
+    firrtl.connect %bundle_vec, %invalid : !firrtl.vector<bundle<a: uint<3>, b: uint<3>>, 3>, !firrtl.vector<bundle<a: uint<3>, b: uint<3>>, 3>
+    %0 = firrtl.subindex %bundle_vec[0] : !firrtl.vector<bundle<a: uint<3>, b: uint<3>>, 3>
+
+    // bundle_vec[0].a <= 4
+    // bundle_vec[0].b <= 5
+    %c4_ui3 = firrtl.constant 4 : !firrtl.uint<3>
+    %c5_ui3 = firrtl.constant 5 : !firrtl.uint<3>
+    %1 = firrtl.subfield %0(0) : (!firrtl.bundle<a: uint<3>, b: uint<3>>) -> !firrtl.uint<3>
+    %2 = firrtl.subfield %0(1) : (!firrtl.bundle<a: uint<3>, b: uint<3>>) -> !firrtl.uint<3>
+    firrtl.connect %1, %c4_ui3 : !firrtl.uint<3>, !firrtl.uint<3>
+    firrtl.connect %2, %c5_ui3 : !firrtl.uint<3>, !firrtl.uint<3>
+
+    // bundle_vec[2] <= bundle_vec[0]
+    %3 = firrtl.subindex %bundle_vec[2] : !firrtl.vector<bundle<a: uint<3>, b: uint<3>>, 3>
+    firrtl.connect %3, %0 : !firrtl.bundle<a: uint<3>, b: uint<3>>, !firrtl.bundle<a: uint<3>, b: uint<3>>
+
+    // res1 <= bundle_vec[2].a = 4
+    // res2 <= bundle_vec[2].b = 5
+    %4 = firrtl.subfield %3(0) : (!firrtl.bundle<a: uint<3>, b: uint<3>>) -> !firrtl.uint<3>
+    %5 = firrtl.subfield %3(1) : (!firrtl.bundle<a: uint<3>, b: uint<3>>) -> !firrtl.uint<3>
+    firrtl.connect %res1, %4 : !firrtl.uint<3>, !firrtl.uint<3>
+    firrtl.connect %res2, %5 : !firrtl.uint<3>, !firrtl.uint<3>
+
+    // TODO: Register and connections are not currently erased.
+    // CHECK: %bundle_vec = firrtl.reg %clock
+    // CHECK: firrtl.subindex %bundle_vec[0] : !firrtl.vector<bundle<a: uint<3>, b: uint<3>>, 3>
+    // CHECK: firrtl.subindex %bundle_vec[2] : !firrtl.vector<bundle<a: uint<3>, b: uint<3>>, 3>
+
+    // Check that results are constant folded.
+    // CHECK:      %[[c4_ui3:.+]] = firrtl.constant 4 : !firrtl.uint<3>
+    // CHECK:      %[[c5_ui3:.+]] = firrtl.constant 5 : !firrtl.uint<3>
+
+    // CHECK:      firrtl.connect %res1, %[[c4_ui3]] : !firrtl.uint<3>, !firrtl.uint<3>
+    // CHECK-NEXT: firrtl.connect %res2, %[[c5_ui3]] : !firrtl.uint<3>, !firrtl.uint<3>
```

Comment:
This type of stuff would be better to put in an XFAIL test.

Reply:
I'm not sure I understand correctly but it doesn't contain failure tests. Could you elaborate?